get-free-epic-games.js

```diff
@@ -7,6 +7,21 @@ const parseFreeGames = function(response) {
   let nextWeekFreeGames = [];
   console.log(JSON.stringify(response.data));
   response.data.data.Catalog.searchStore.elements.forEach(game => {
+    let slug = null
+
+    try {
+      slug = game.catalogNs.mappings[0].pageSlug
+      
+      // fallback to old implementation if this fails
+      if (!slug || slug == "null") {
+        slug = game.productSlug
+      }
+
+    } catch(e) {
+      slug = game.productSlug
+    }
```

Comment:
Not sure if there's a cleaner way to do this with TypeErrors. Typically in js we can just do

```
slug = game.catalogNs.mappings[0].pageSlug
if (!slug || slug == null) {
   slug = game.productSlug
}
```

But the first line causes a TypeError, presumably because we're enforcing types in the project as a setting somewhere?

Reply:
I have no idea. My javascript is not good enough. I thought there are no types unless you use typescript. I'm happy that it works.